src/Castle.Core/DynamicProxy/Generators/Emitters/ClassEmitter.cs

```diff
@@ -66,6 +66,11 @@ public ModuleScope ModuleScope
 			get { return moduleScope; }
 		}
 
+		internal bool InStrongNamedModule
+		{
+			get { return StrongNameUtil.IsAssemblySigned(TypeBuilder.Assembly); }
+		}
```

Comment:
This property gets checked for every invocation type being generated, so this *might* have a slight performance impact. I guess the result returned by `StrongNameUtil.IsAssemblySigned` could be cached here, in `ClassEmitter`. Seeing that `StrongNameUtil` already performs caching, all we'd really save at this point is the time it takes to enter and leave the `lock` inside `StrongNameUtil`. And caching here might actually introduce a `lock` of its own, so just going ahead and performing this call instead of caching the result is probably good enough.